aws-codeartifact-repository/src/main/java/software/amazon/codeartifact/repository/UpdateHandler.java

```diff
@@ -22,88 +32,145 @@
+            .then(progress -> {
+                Set<String> externalConnectionsToRemove = Sets.difference(currentExternalConnections, requestedExternalConnections);
+                return disassociateExternalConnections(progress, callbackContext, request, proxyClient, externalConnectionsToRemove, logger);
+            })
+            .then(progress -> {
+                Set<String> externalConnectionsToAdd = Sets.difference(requestedExternalConnections, currentExternalConnections);
+                return associateExternalConnections(progress, callbackContext, request, proxyClient, externalConnectionsToAdd, logger);
+            })
+            .then(progress -> updateRepository(proxy, progress, callbackContext, proxyClient, logger))
             .then(progress -> new ReadHandler().handleRequest(proxy, request, callbackContext, proxyClient, logger));
     }
+
+    private ProgressEvent<ResourceModel, CallbackContext> updateRepository(
+        final AmazonWebServicesClientProxy proxy,
+        final ProgressEvent<ResourceModel, CallbackContext> progress,
+        final CallbackContext callbackContext,
+        final ProxyClient<CodeartifactClient> proxyClient, Logger logger
+    ) {
+        return proxy.initiate("AWS-CodeArtifact-Repository::Create", proxyClient,progress.getResourceModel(), callbackContext)
+            .translateToServiceRequest(Translator::translateToUpdateRepository)
+            .makeServiceCall((awsRequest, client) -> {
+                AwsResponse awsResponse = null;
+                try {
+                    awsResponse = client.injectCredentialsAndInvokeV2(awsRequest, client.client()::updateRepository);
+                } catch (final AwsServiceException e) {
+                    String repositoryName = progress.getResourceModel().getRepositoryName();
+                    Translator.throwCfnException(e, Constants.CREATE_REPOSITORY, repositoryName);
+                }
+                logger.log(String.format("%s successfully created.", ResourceModel.TYPE_NAME));
+                return awsResponse;
+            })
+            .progress();
+    }
+
+    protected ProgressEvent<ResourceModel, CallbackContext> updateRepositoryPermissionsPolicy(
+        final AmazonWebServicesClientProxy proxy,
+        final ProgressEvent<ResourceModel, CallbackContext> progress,
+        final CallbackContext callbackContext,
+        final ResourceHandlerRequest<ResourceModel> request,
+        final ProxyClient<CodeartifactClient> proxyClient,
+        final Logger logger
+    ) {
+        final ResourceModel desiredModel = progress.getResourceModel();
+        if (desiredModel.getPolicyDocument() != null) {
+            return putRepositoryPermissionsPolicy(proxy, progress, callbackContext, proxyClient, logger);
+        }
+        return deleteRepositoryPermissionsPolicy(proxy, progress, callbackContext, proxyClient, logger);
```

Comment:
Should we be checking to see what the current policy is and only updating if it changed? And similarly, should we only delete if there was a policy?

Reply:
Yes that'd would prevent unnecessary calls, I'll add in a check